### PR TITLE
Claim hooks 2 - ready to fetch

### PR DESCRIPTION
**Pull request by alfetopito**
# Summary

Adapted fetch functions to expected document format coming from future claims repo

- Modded `useUserClaims`
- Using fake data but at a different stage
- Converting string types to enum types (`Airdrop` -> `0`)
- Renamed button `Claim vCOW` to `vCOW`
- Made `vCOW` button always visible

  # To Test

1. Connect wallet on rinkeby
2. Click on `vCOW` button
3. Click on claim
4. Execute tx
* Claim will be performed and claimable amount will be reduced

**Note**: Each claim can be done only once, so if you can't claim, try using a different claim index in this file
Screenshot from 2021-12-24 17-08-53

  # Background

*Optional: Give background information for changes you've made, that might be difficult to explain via comments*

### CODE REVIEW

**Review by fedgiac — COMMENTED** [review]
I tried the test plan and after I connected my Metamask it worked, [here](https://rinkeby.etherscan.io/tx/0xeefdb9640da952e28fe61126a20d3c24c2f3e631a16d408bdb5738cdbd1699a4) is the transaction. :heavy_check_mark:

I can't say much about how React-specific things are used, but everything contract related looks good to me.

File: src/custom/state/claim/hooks/index.ts [hunk 1]
```diff
@@ -28,6 +31,19 @@ export const enum ClaimType {
   Advisor, // free, with vesting, only on mainnet
 }
 
+// TODO: find a way (if possible) to get this from the ClaimType enum
+type RepoClaimType = 'Airdrop' | 'GnoOption' | 'UserOption' | 'Investor' | 'Team' | 'Advisor'
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/state/claim/hooks/index.ts:35]: Ahhhh nice, thanks for this!

File: src/custom/state/claim/hooks/index.ts [hunk 2]
```diff
@@ -28,6 +31,19 @@ export const enum ClaimType {
...
+
+// TODO: also, is there a smarter way of doing this?
+export const REVERSE_CLAIM_TYPE_MAPPING: Record<RepoClaimType, ClaimType> = {
+  Airdrop: ClaimType.Airdrop,
+  GnoOption: ClaimType.GnoOption,
+  UserOption: ClaimType.UserOption,
+  Investor: ClaimType.Investor,
+  Team: ClaimType.Team,
+  Advisor: ClaimType.Advisor,
+}
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/state/claim/hooks/index.ts:45]: Awesome tip, missed it completely 🤦

File: src/custom/state/claim/hooks/hooksMod.ts [hunk 3]
```diff
@@ -26,34 +27,80 @@ export { useUserClaimData } from '@src/state/claim/hooks'
...
+  //   FETCH_CLAIM_FILE_PROMISES[key] ??
+  //   (FETCH_CLAIM_FILE_PROMISES[key] = fetch(`${CLAIMS_REPO}${key}.json`) // mod
+  //     .then((res) => res.json())
+  //     .catch((error) => {
+  //       console.error(`Failed to get claim file mapping for starting address ${key}`, error)
+  //       delete FETCH_CLAIM_FILE_PROMISES[key]
+  //     }))
+  // )
+  return Promise.resolve({
+    '0x0010B775429d6C92333E363CBd6BF28dDF1A87E6': [
```
[the remark this answers is not in the document]
Comment (alfetopito) [reply src/custom/state/claim/hooks/hooksMod.ts:57]: Yep, I am aware. Just for the sake of testing I copied only one

**Comment by alfetopito** [comment]
FYI, I rebased with latest from `claim` branch which includes Michel's components, thus the claim action no longer works.
Since Federico tested and it's working, I'll merge as is.
Feel free to add comments afterwards and I'll address them in follow up PRs.

**Pull request merged**